### ISSUE TITLE
Improve iteration estimation in LightGBM

### ISSUE THREAD

**Issue by paxcema** [mindsdb/lightwood#404]
Currently, the policy we have to engage Optuna in the LightGBM mixer can be too optimistic at times. This can lead to very slow hyperparameter optimization phases. We should improve the estimated time so that the mixer is generally faster without losing (much) accuracy.

**Comment by paxcema** [comment]
Added bug label because the current implementation can go on for longer than the specified maximum time budget
